### PR TITLE
Add sign extension spec tests; add sign extension to singlepass

### PR DESCRIPTION
**Pull request by MarkMcCaskey**
Adds missing spectests from official repo, implements instructions for singlepass

# Review

- [ ] Add a short description of the the change to the CHANGELOG.md file

### CODE REVIEW

**Comment by syrusakbary** [comment]
Nice!

**Comment by nlewycky** [comment]
Sign extension ops are pre-standard still? Update the PrestandardFeatures struct in src/bin/wasmer.rs and propagate the flag through from the runner to the backends. Should be easy to search "simd" and follow the same pattern.

**Comment by MarkMcCaskey** [comment]
> Sign extension ops are pre-standard still? Update the PrestandardFeatures struct in src/bin/wasmer.rs and propagate the flag through from the runner to the backends. Should be easy to search "simd" and follow the same pattern.

They are, they're already implemented in LLVM and Clif though and sounds good!

**Comment by MarkMcCaskey** [comment]
bors try

File: src/bin/wasmer.rs [hunk 1]
```diff
@@ -91,11 +91,41 @@ struct PrestandardFeatures {
     #[structopt(long = "enable-threads")]
     threads: bool,
 
+    /// Enable support for the threads proposal.
```
Comment (losfair) [inline src/bin/wasmer.rs]: threads -> sign extension?

File: lib/singlepass-backend/src/codegen_x64.rs [hunk 2]
```diff
@@ -2948,6 +2949,125 @@ impl FunctionCodeGenerator<CodegenError> for X64FunctionCode {
...
+                let ret = self.machine.acquire_locations(
+                    a,
+                    &[(WpType::I32, MachineValue::WasmStack(self.value_stack.len()))],
+                    false,
+                )[0];
+                self.value_stack.push(ret);
+
+                let tmpg1 = self.machine.acquire_temp_gpr().unwrap();
+                a.emit_mov(Size::S32, loc, Location::GPR(tmpg1));
+                a.emit_and(Size::S32, Location::Imm32(0xFF), Location::GPR(tmpg1));
```
Comment (losfair) [inline lib/singlepass-backend/src/codegen_x64.rs]: Do we really need to `and` the value with a corresponding bitmask first, while `mov{zx/sx}` is aware of the width?

Comment (MarkMcCaskey) [reply lib/singlepass-backend/src/codegen_x64.rs]: Good call!  It seems that we don't!

**Comment by MarkMcCaskey** [comment]
bors try

**Comment by MarkMcCaskey** [comment]
bors try-

**Comment by MarkMcCaskey** [comment]
bors try

**Pull request merged**